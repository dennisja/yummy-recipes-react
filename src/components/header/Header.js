import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import ProfileCard from '../profile/ProfileCard';

export const MenuItems = (props) => {
    const {
            loggedIn, logoutUser, userName, activeClass,
          } = props;
    if (loggedIn) {
        return (
          <div>
            <li>
              <NavLink to={loggedIn ? '/home' : '/'} activeClassName={activeClass} >
                <i className="fa fa-home" /> Home
              </NavLink>
            </li>
            <li>
              <NavLink to="/recipes" activeClassName={activeClass}>
                <i className="fa fa-cutlery" /> Recipes
              </NavLink>
            </li>
            <li>
              <NavLink to="/categories" activeClassName={activeClass}>
                <i className="fa fa-tags" /> Categories
              </NavLink>
            </li>
            <li>
              <NavLink
                to="/profile"
                activeClassName={activeClass}
                id="pdrop"
              >
                <i className="fa fa-user" /> {userName}
              </NavLink>
            </li>
            <li>
              <NavLink
                to="/"
                onClick={logoutUser}
              > <i className="fa fa-sign-out" /> Logout
              </NavLink>
            </li>
          </div>);
    }
    return null;
};

MenuItems.propTypes = {
    loggedIn: PropTypes.bool.isRequired,
    logoutUser: PropTypes.func.isRequired,
    userName: PropTypes.string.isRequired,
    activeClass: PropTypes.string,
};

MenuItems.defaultProps = {
  activeClass: 'active-nav-link',
};

const Header = props => (
  <header>
    <nav>
      <div className="nav-wrapper orange">
        <NavLink to={props.loggedIn ? '/home' : '/'} className="brand-logo">
        Yummy Recipes
        </NavLink>
        <NavLink to="/" data-activates="mobile-demo" className="button-collapse">
          <i className="fa fa-bars" />
        </NavLink>
        <ul className="right hide-on-med-and-down" >
          <MenuItems
            loggedIn={props.loggedIn}
            logoutUser={props.logoutUser}
            userName={props.userData ? props.userData.firstname : 'Profile'}
          />
        </ul>
        <ul className="side-nav" id="mobile-demo">
          {/* render the profile preview on mobile */}
          {
            props.loggedIn
            ? <ProfileCard userData={props.userData} />
            : null
          }
          <MenuItems
            loggedIn={props.loggedIn}
            logoutUser={props.logoutUser}
            userName={props.userData ? props.userData.firstname : 'Profile'}
            activeClass="active-sidebar-link"
          />
        </ul>
      </div>
    </nav>
  </header>
);

Header.propTypes = {
    loggedIn: PropTypes.bool.isRequired,
    logoutUser: PropTypes.func.isRequired,
    userData: PropTypes.object,
};

Header.defaultProps = {
  userData: null,
};

export default Header;
