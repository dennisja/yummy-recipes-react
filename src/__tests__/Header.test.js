import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header, { MenuItems } from '../components/header/Header';

// mocking react dom
// jest.mock('react-dom');
const logoutUser = jest.fn();

describe('Test Header component', () => {
    const userData = { firstname: 'D', lastname: 'E', id: '1' };
    const headerOuput = shallow(<Header loggedIn logoutUser={logoutUser} userData={userData} />);
    const headerLoggedOuOuput = shallow(<Header loggedIn={false} logoutUser={logoutUser} />);

    it('renders correctly when logged in', () => {
        expect(shallowToJson(headerOuput)).toMatchSnapshot();
    });

    it('renders correctly when logged out', () => {
        expect(shallowToJson(headerLoggedOuOuput)).toMatchSnapshot();
    });

    it('has two NavLinks', () => {
        expect(headerOuput.find('NavLink')).toHaveLength(2);
    });

    it('renders a ProfileCard component', () => {
        expect(headerOuput.find('ProfileCard')).toHaveLength(1);
    });

    it('renders different content when loggedin and out', () => {
        expect(headerOuput === headerLoggedOuOuput).toBe(false);
    });
});

describe('Test MenuItems comonent ', () => {
    const menuItemsOutput = shallow(<MenuItems loggedIn logoutUser={logoutUser} userName="D" />);
    it('renders correctly', () => {
        expect(shallowToJson(menuItemsOutput)).toMatchSnapshot();
    });

    it('has five NavLinks when logged in', () => {
        expect(menuItemsOutput.find('NavLink').length).toBe(5);
    });

    it('renders nothing when a user is logged out', () => {
        expect(menuItemsOutput.find('NavLink').length).toBe(5);
        menuItemsOutput.setProps({ loggedIn: false });
        expect(menuItemsOutput.find('NavLink')).toHaveLength(0);
        expect(menuItemsOutput.equals(null)).toBe(true);
    });
});

