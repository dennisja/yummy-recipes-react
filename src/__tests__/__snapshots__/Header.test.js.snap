// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component renders correctly when logged in 1`] = `
<header>
  <nav>
    <div
      className="nav-wrapper orange"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        className="brand-logo"
        to="/home"
      >
        Yummy Recipes
      </NavLink>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        className="button-collapse"
        data-activates="mobile-demo"
        to="/"
      >
        <i
          className="fa fa-bars"
        />
      </NavLink>
      <ul
        className="right hide-on-med-and-down"
      >
        <Component
          activeClass="active-nav-link"
          loggedIn={true}
          logoutUser={[Function]}
          userName="D"
        />
      </ul>
      <ul
        className="side-nav"
        id="mobile-demo"
      >
        <ProfileCard
          userData={
            Object {
              "firstname": "D",
              "id": "1",
              "lastname": "E",
            }
          }
        />
        <Component
          activeClass="active-sidebar-link"
          loggedIn={true}
          logoutUser={[Function]}
          userName="D"
        />
      </ul>
    </div>
  </nav>
</header>
`;

exports[`Test Header component renders correctly when logged out 1`] = `
<header>
  <nav>
    <div
      className="nav-wrapper orange"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        className="brand-logo"
        to="/"
      >
        Yummy Recipes
      </NavLink>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        className="button-collapse"
        data-activates="mobile-demo"
        to="/"
      >
        <i
          className="fa fa-bars"
        />
      </NavLink>
      <ul
        className="right hide-on-med-and-down"
      >
        <Component
          activeClass="active-nav-link"
          loggedIn={false}
          logoutUser={[Function]}
          userName="Profile"
        />
      </ul>
      <ul
        className="side-nav"
        id="mobile-demo"
      >
        <Component
          activeClass="active-sidebar-link"
          loggedIn={false}
          logoutUser={[Function]}
          userName="Profile"
        />
      </ul>
    </div>
  </nav>
</header>
`;

exports[`Test MenuItems comonent  renders correctly 1`] = `
<div>
  <li>
    <NavLink
      activeClassName="active-nav-link"
      ariaCurrent="true"
      to="/home"
    >
      <i
        className="fa fa-home"
      />
       Home
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active-nav-link"
      ariaCurrent="true"
      to="/recipes"
    >
      <i
        className="fa fa-cutlery"
      />
       Recipes
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active-nav-link"
      ariaCurrent="true"
      to="/categories"
    >
      <i
        className="fa fa-tags"
      />
       Categories
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active-nav-link"
      ariaCurrent="true"
      id="pdrop"
      to="/profile"
    >
      <i
        className="fa fa-user"
      />
       
      D
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      onClick={[Function]}
      to="/"
    >
       
      <i
        className="fa fa-sign-out"
      />
       Logout
    </NavLink>
  </li>
</div>
`;
